http-agent/src/main/java/org/motechproject/http/agent/listener/HttpClientEventListener.java

```diff
@@ -118,24 +122,79 @@ public void handle(MotechEvent motechEvent) {
+     * parameters. Returns the response from the performed request. Retry count(default value is 1) and retry
+     * interval(default value is 0, expressed in milliseconds) can be specified by event parameters(keys:
+     * org.motechproject.http.agent.domain.EventDataKeys.RETRY_COUNT and org.motechproject.http.agent.domain.EventDataKeys.RETRY_INTERVAL).
+     *
+     * @param motechEvent the event which contains data for request
+     * @return response from the posted request
+     */
+    @MotechListener (subjects = { SendRequestConstants.SEND_REQUEST_SUBJECT })
+    public ResponseEntity<?> handleWithUserPasswordAndReturnType(MotechEvent motechEvent) {
+        Map<String, Object> parameters = motechEvent.getParameters();
+        final String url = String.valueOf(parameters.get(SendRequestConstants.URL));
+        Object requestData = parameters.get(SendRequestConstants.DATA);
+        String username = String.valueOf(parameters.get(SendRequestConstants.USERNAME));
+        String password = String.valueOf(parameters.get(SendRequestConstants.PASSWORD));
+        boolean redirection = (boolean) parameters.get(SendRequestConstants.REDIRECTION_ABILTY);
+        LOGGER.info(String.format("Posting Http request -- Url: %s, Data: %s",
+                url, String.valueOf(requestData)));
+
+        HttpEntity<Object> entity = new HttpEntity<>(requestData);
+
+        RestTemplate restTemplate;
+
+        if (StringUtils.isNotBlank(username)
+                || StringUtils.isNotBlank(password)) {
+            restTemplate = new RestTemplate(usernamePasswordRequestFactory(
+                    username, password));
+        } else {
+            restTemplate = basicRestTemplate;
+        }
+        
+        if (!redirection) {
+            final HttpComponentsClientHttpRequestFactory factory = new HttpComponentsClientHttpRequestFactory();
+            HttpClient httpClient = HttpClientBuilder.create().disableRedirectHandling().build();
+            factory.setHttpClient(httpClient);
+            restTemplate.setRequestFactory(factory);
+        }
+
+        int retryCount = 1; // default retry count = 1
+        long retryInterval = 0; // by default, no waiting time between two
+        // retries
+        if (parameters.get(EventDataKeys.RETRY_COUNT) != null) {
+            retryCount = (int) parameters.get(EventDataKeys.RETRY_COUNT);
+        }
+        if (parameters.get(EventDataKeys.RETRY_INTERVAL) != null) {
+            retryInterval = (long) parameters.get(EventDataKeys.RETRY_INTERVAL);
+        }
+
+        return checkStatusCodeAndReturnResponse(url, entity, Method.POST, restTemplate, retryCount, retryInterval);
+    }
+
+    private ResponseEntity<?> checkStatusCodeAndReturnResponse(String url, Object requestData, Method method, RestTemplate restTemplate, int retryCount, long retryInterval)
```

Comment:
This method name seems a little bit out of place. Aren't we simply sending the request here? Could use more appropriate name.